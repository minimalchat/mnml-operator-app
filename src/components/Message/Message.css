.Message__client {
  align-self: flex-end;
  background: #ffffff;
  border-radius: 10px 10px 0 10px;
  color: #000;
  float: left;
  font-size: 12px;
  letter-spacing: 1px;
  list-style: none;
  margin-top: 6px;
  margin-right: 6px;
  max-width: 45%;
  min-width: 25%;
  padding: 10px;
  text-align: left;
}

.Message__client ul li {
  text-align: right;
}

.Message__client.typing {
  padding: 3px 10px 3px 10px;
  min-width: auto;
}

.Message__operator {
  align-self: flex-start;
  background: #0a6bef;
  border-radius: 10px 10px 10px 0;
  color: white;
  font-size: 12px;
  letter-spacing: 1px;
  list-style: none;
  padding: 0.5rem;
  margin: 0.5rem;
  max-width: 45%;
  min-width: 25%;
}

.Message__operator ul,
.Message__client ul {
  list-style: none;
  padding: 0;
  margin: 0;
}

.Message__operator ul li {
  text-align: left;
}
