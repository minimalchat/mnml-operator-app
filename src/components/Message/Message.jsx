/**
  * @summary: Displays a single message in the Message List
*/

import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import './Message.css';

const Message = (props) => {
  let { content } = props;
  const { author, timestamp } = props;
  const msgClass = () => (
    author.indexOf('client') >= 0 ? 'Message__client' : 'Message__operator'
  );

  content = content.map((message, index) => <li key={`${author}`}>{message}</li>);

  let datetime = null;

  // Only show the timestamp on client messages
  if (author.indexOf('client') >= 0) {
    const momentTimestamp = moment(timestamp);

    // If the timestamp is less than 24 hours, show relative time, otherwise
    //  show Day of week, Time
    datetime = (
      <span className="Message__time">
        {
          momentTimestamp.isBefore(moment(), 'day')
            ? momentTimestamp.format('ddd, h:mma')
            : momentTimestamp.fromNow()
        }
      </span>
    );
  }

  let message = (
    <div>
      <ul>
        {content}
      </ul>
      {datetime}
    </div>
  );

  return (
    <li className={msgClass()} style={{ clear: 'both' }}>
      {message}
    </li>
  );
};

Message.propTypes = {
  author: PropTypes.string.isRequired,
  timestamp: PropTypes.string.isRequired,
  content: PropTypes.arrayOf(
    PropTypes.string,
  ),
};

Message.defaultProps = {
  content: [],
};


export default Message;

