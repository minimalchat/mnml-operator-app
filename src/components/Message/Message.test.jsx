
import React from 'react';
import { shallow } from 'enzyme';
import Message from './Message.jsx';

/*
const store = {
  subscribe: jest.fn(),
  dispatch: jest.fn(),
  getState: jest.fn(() => ({})),
};
*/


describe('MessageList', () => {
  it('matches snapshot', () => {
    const component = shallow(<Message author="operator.steve" content={['test message']} timestamp={new Date().toString()} />);
    expect(component).toMatchSnapshot();
  });
});
