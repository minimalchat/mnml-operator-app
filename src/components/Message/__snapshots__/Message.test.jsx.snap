// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList matches snapshot 1`] = `
<li
  className="Message__operator"
  style={
    Object {
      "clear": "both",
    }
  }
>
  <div>
    <ul>
      <li
        key="operator.steve"
      >
        test message
      </li>
    </ul>
  </div>
</li>
`;
